admin_ui_support/modules/jsonapi_tables/src/ReadOnlyTableEntityStorage.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Drupal\jsonapi_tables;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\Sql\SqlContentEntityStorage;
+
+/**
+ * SQL Entity storage for entity types that wrap readonly tables.
+ */
+class ReadOnlyTableEntityStorage extends SqlContentEntityStorage {
+
+  use EntityTypeTableSchemaTrait;
+  /**
+   * {@inheritdoc}
+   */
+  public function loadByProperties(array $values = []) {
+    // @todo Can we add uuid to tabls to remove this hack?
```

Comment:
It feels like it could be useful to have that in dblog entries themselves. Maybe we could open up a core issue about that?

Reply:
I don't really see that watchdog entries need to be "Universally Unique across multiple systems"

I do think where this exposing of the tables falls down in what Drupal entities are meant to be since UUID are required for Drupal entities.

So maybe this would make incompatible with some parts of core. Workspaces? 

So I think by time we want to add this kind of functionality to core might make sense make changes to jsonapi to allow it to expose non-entity things instead of this method.